RequestKit/Router.swift

```diff
@@ -119,7 +119,11 @@ public extension Router {
         let task = session.dataTask(with: request) { data, response, err in
             if let response = response as? HTTPURLResponse {
                 if response.wasSuccessful == false {
-                    let error = NSError(domain: self.configuration.errorDomain, code: response.statusCode, userInfo: nil)
+                    var userInfo = [String: AnyObject]()
+                    if let data = data, let json = try? JSONSerialization.jsonObject(with: data, options: .mutableContainers) as? [String: AnyObject] {
+                        userInfo["json"] = json as AnyObject?
```

Comment:
I would use an explicit constant public key like `public let RequestKitJSONErrorKey = "RequestKitJSONErrorKey"` and set this instead of just `json`

Reply:
I was hoping to also make use of `NSError:ErrorUserInfoKey
` but I still have to play around with it for a bit to see which is best.

Reply:
We could probably also use the already existing `public let RequestKitErrorResponseKey = "RequestKitErrorResponseKey"` for the JSON key.

Reply:
Yep